<a href="https://colab.research.google.com/github/SLebriz2002/gestion_des_evenements/blob/main/notebooks/controle_numpy_matplotlib.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Contrôle 01 — Data Science avec NumPy et Matplotlib
**Master 1 — Intelligence Artificielle & Génie Informatique (FPK / USMS)**

Nom & Prénom : Ismahane Lebriz  
Filière :  Systeme d'information et intelligence artificielle
Année universitaire : 2025 / 2026

---

**Règles :**
- Bibliothèques autorisées : NumPy, Matplotlib
- Toute autre bibliothèque est interdite
- Chaque question doit être traitée dans une cellule dédiée
- Chaque résultat doit être brièvement interprété


# Partie A — Importation du Dataset, Arrays, Premières Manipulations

### A1. Importation du dataset
Importez NumPy et Matplotlib.  
Chargez le fichier `iris.csv` avec `np.genfromtxt` (`delimiter=','`, `skip_header=1`).  
Stockez les données dans un array nommé `data` et affichez sa `shape`.


In [5]:
# Votre code ici
import numpy as np
import matplotlib.pyplot as plt
data = np.genfromtxt('Iris.csv' , delimiter = ',' , skip_header = 1)
print(data.shape)



(150, 6)


j'ai importe les bibliotheque Numpy et Matplotlib puis j'ai charge les donnees iris.csv avec np.genfromtxt en les stockant dans un array nomme data enfin j'ai affiche sa tuple des tailles par dimensions.
j'ai trouve 150 ligne et 6 colonnes


### A2. Séparation des features et des espèces
Séparez :
- les 4 premières colonnes numériques dans un array `features`
- la dernière colonne (espèces) dans un array `species`
Utilisez le slicing.


In [20]:
# Votre code ici
features = data[:, :4]
species = data[: , 4]
print("fetures :\n ",features)
print("species  :\n ",species)
print("fetures shape :\n ",features.shape)
print("species shape :\n ",species.shape)

fetures :
  [[  1.    5.1   3.5   1.4]
 [  2.    4.9   3.    1.4]
 [  3.    4.7   3.2   1.3]
 [  4.    4.6   3.1   1.5]
 [  5.    5.    3.6   1.4]
 [  6.    5.4   3.9   1.7]
 [  7.    4.6   3.4   1.4]
 [  8.    5.    3.4   1.5]
 [  9.    4.4   2.9   1.4]
 [ 10.    4.9   3.1   1.5]
 [ 11.    5.4   3.7   1.5]
 [ 12.    4.8   3.4   1.6]
 [ 13.    4.8   3.    1.4]
 [ 14.    4.3   3.    1.1]
 [ 15.    5.8   4.    1.2]
 [ 16.    5.7   4.4   1.5]
 [ 17.    5.4   3.9   1.3]
 [ 18.    5.1   3.5   1.4]
 [ 19.    5.7   3.8   1.7]
 [ 20.    5.1   3.8   1.5]
 [ 21.    5.4   3.4   1.7]
 [ 22.    5.1   3.7   1.5]
 [ 23.    4.6   3.6   1. ]
 [ 24.    5.1   3.3   1.7]
 [ 25.    4.8   3.4   1.9]
 [ 26.    5.    3.    1.6]
 [ 27.    5.    3.4   1.6]
 [ 28.    5.2   3.5   1.5]
 [ 29.    5.2   3.4   1.4]
 [ 30.    4.7   3.2   1.6]
 [ 31.    4.8   3.1   1.6]
 [ 32.    5.4   3.4   1.5]
 [ 33.    5.2   4.1   1.5]
 [ 34.    5.5   4.2   1.4]
 [ 35.    4.9   3.1   1.5]
 [ 36.    5.    3.2   1.2]
 [ 37.    5.5   

Interpretation : apres la separation des colonnes ,
la dimension des features est (150,4)
la dimension des species est (150,)

### A3. Indexing
Affichez les 5 premières lignes de `features` en utilisant l’indexing.


In [12]:
# Votre code ici
print("les cinq premieres lignes de features :\n" , features[:5])

les cinq premieres lignes de features :
 [[1.  5.1 3.5 1.4]
 [2.  4.9 3.  1.4]
 [3.  4.7 3.2 1.3]
 [4.  4.6 3.1 1.5]
 [5.  5.  3.6 1.4]]


affichage de 5 premieres ligne de features

### A4. Extraction d’une colonne
Créez un array `sepal_length` contenant uniquement la première colonne de `features`.  
Affichez sa `size` et son `dtype`.


In [13]:
# Votre code ici
sepal_length = features[: , 0]
print("size of sepal_length", sepal_length.size)
print("dtype of sepal_length", sepal_length.dtype)

size of sepal_length 150
dtype of sepal_length float64


interpretation : sepal_lenght contient 150 valeurs de type float64


### A5. Slicing par espèce
Utilisez le slicing pour extraire les 50 premières lignes correspondant à l’espèce *Setosa*.  
Stockez-les dans `setosa_features`.


In [14]:
# Votre code ici
setosa_features = features[:50]
print(setosa_features)
print("shape of setosa ", setosa_features.shape)

[[ 1.   5.1  3.5  1.4]
 [ 2.   4.9  3.   1.4]
 [ 3.   4.7  3.2  1.3]
 [ 4.   4.6  3.1  1.5]
 [ 5.   5.   3.6  1.4]
 [ 6.   5.4  3.9  1.7]
 [ 7.   4.6  3.4  1.4]
 [ 8.   5.   3.4  1.5]
 [ 9.   4.4  2.9  1.4]
 [10.   4.9  3.1  1.5]
 [11.   5.4  3.7  1.5]
 [12.   4.8  3.4  1.6]
 [13.   4.8  3.   1.4]
 [14.   4.3  3.   1.1]
 [15.   5.8  4.   1.2]
 [16.   5.7  4.4  1.5]
 [17.   5.4  3.9  1.3]
 [18.   5.1  3.5  1.4]
 [19.   5.7  3.8  1.7]
 [20.   5.1  3.8  1.5]
 [21.   5.4  3.4  1.7]
 [22.   5.1  3.7  1.5]
 [23.   4.6  3.6  1. ]
 [24.   5.1  3.3  1.7]
 [25.   4.8  3.4  1.9]
 [26.   5.   3.   1.6]
 [27.   5.   3.4  1.6]
 [28.   5.2  3.5  1.5]
 [29.   5.2  3.4  1.4]
 [30.   4.7  3.2  1.6]
 [31.   4.8  3.1  1.6]
 [32.   5.4  3.4  1.5]
 [33.   5.2  4.1  1.5]
 [34.   5.5  4.2  1.4]
 [35.   4.9  3.1  1.5]
 [36.   5.   3.2  1.2]
 [37.   5.5  3.5  1.3]
 [38.   4.9  3.1  1.5]
 [39.   4.4  3.   1.3]
 [40.   5.1  3.4  1.5]
 [41.   5.   3.5  1.3]
 [42.   4.5  2.3  1.3]
 [43.   4.4  3.2  1.3]
 [44.   5. 

interpretation : le code affiche 50 ligne et 4 colonnes c est a dire 50 fleurs setosa et chacune fleur a 4 caracteristiques

# Partie B — Statistiques Basiques et Opérations Mathématiques

### B6. Moyenne par colonne
Calculez la moyenne de chaque colonne de `features` avec `np.mean(axis=0)`.


In [18]:
# Votre code ici
moy = np.mean(features , axis = 0)
print("la moyenne  de chaque colonne est : \n" , moy)

la moyenne  de chaque colonne est : 
 [75.5         5.84333333  3.054       3.75866667]


la moyenne de chaque colonnes dans le array features


### B7. Min et Max
Trouvez la valeur minimale et maximale de la longueur des pétales.


In [ ]:
# Votre code ici
petal_lenghth = features[:,2]

### B9. Normalisation
Normalisez les longueurs des pétales en les divisant par leur maximum.
Stockez le résultat dans `normalized_petals`.


In [ ]:
# Votre code ici


### B10. Somme cumulée
Calculez la somme cumulée des largeurs des pétales pour Versicolor.
Affichez les 10 premières valeurs.


In [ ]:
# Votre code ici


### B10. Somme cumulée
Calculez la somme cumulée des largeurs des pétales pour Versicolor.
Affichez les 10 premières valeurs.


In [ ]:
# Votre code ici


# Partie C — Transformations Avancées avec NumPy

### C11. Espèces uniques
Listez les espèces uniques avec `np.unique` et comptez leurs occurrences.


In [ ]:
# Votre code ici


### C12. Reshape par espèce
Transformez `features` en un array de forme `(3, 50, 4)` nommé `features_by_species`.


In [ ]:
# Votre code ici


### C13. Concaténation
Concaténez horizontalement les moyennes par colonne avec un array de zéros.


In [ ]:
# Votre code ici


### C14. Data cleaning (simulation)
Simulez des valeurs NaN et remplacez-les avec `np.nan_to_num`.


In [ ]:
# Votre code ici


### C15. Broadcasting
Ajoutez 1 à toutes les valeurs de la première colonne de `features`.


In [ ]:
# Votre code ici


### C16. np.where
Créez un array binaire indiquant si la longueur des pétales est supérieure à 4.0.


In [ ]:
# Votre code ici


# Partie D — Visualisation avec Matplotlib (Plots Simples)

### D17. Line plot
Line plot des longueurs des sépales pour les 50 premières fleurs.


In [ ]:
# Votre code ici


### D18. Scatter plot
Scatter plot : longueur vs largeur des pétales.


In [ ]:
# Votre code ici


### D19. Bar chart
Bar chart des moyennes des quatre caractéristiques.


In [ ]:
# Votre code ici


### D20. Histogram
Histogramme de la largeur des sépales (10 bins).


In [ ]:
# Votre code ici


### D21. Boxplot par espèce
Boxplot des longueurs des pétales par espèce.


In [ ]:
# Votre code ici


# Partie E — Visualisations Avancées

### E22. Subplots comparatifs
Deux scatter plots côte à côte : Setosa vs Versicolor.


In [ ]:
# Votre code ici


### E23. Grille 2x2
Histogrammes des 4 caractéristiques avec `plt.subplots(2,2)`.


In [ ]:
# Votre code ici


### E24. Bar chart avec error bars
Moyennes par espèce avec écart-type.


In [ ]:
# Votre code ici


### E25. Scatter coloré par espèce
Scatter plot avec couleurs par espèce.


In [ ]:
# Votre code ici


### E26. Line plot superposé
Cumsum des largeurs des pétales par espèce.


In [ ]:
# Votre code ici


# Partie F — Mini-Analyse Finale

### F27. Corrélations
Calculez la matrice de corrélation avec `np.corrcoef`.


In [ ]:
# Votre code ici


### F28. Visualisation des corrélations
Visualisez les corrélations sous forme de bar chart ou scatter.


In [ ]:
# Votre code ici


### F29. Variance par espèce
Identifiez l’espèce avec la plus grande variance globale et visualisez ses distributions.


In [ ]:
# Votre code ici


### F30. Conclusion
Rédigez une conclusion contenant **3 insights** issus de votre analyse.


1) ...
2) ...
3) ...